airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/source.py

```diff
@@ -50,17 +52,35 @@ def spec(self, *args, **kwargs) -> ConnectorSpecification:
             supportsIncremental=True,
             supported_destination_sync_modes=[DestinationSyncMode.overwrite, DestinationSyncMode.append, DestinationSyncMode.append_dedup],
             connectionSpecification=SourceTiktokMarketingSpec.schema(),
+            additionalProperties=True,
+            advanced_auth=AdvancedAuth(
+                auth_flow_type=AuthFlowType.oauth2_0,
+                predicate_key=["credentials", "auth_type"],
+                predicate_value="oauth2.0",
+                oauth_config_specification=OAuthConfigSpecification(
+                    complete_oauth_output_specification=CompleteOauthOutputSpecification.schema(),
+                    complete_oauth_server_input_specification=CompleteOauthServerInputSpecification.schema(),
+                    complete_oauth_server_output_specification=CompleteOauthServerOutputSpecification.schema(),
+                ),
+            ),
         )
 
     @staticmethod
     def _prepare_stream_args(config: Mapping[str, Any]) -> Mapping[str, Any]:
         """Converts an input configure to stream arguments"""
+        credentials = config.get("credentials") or config
+
+        access_token = credentials["access_token"]
+        secret = credentials.get("secret") or credentials.get("environment", {}).get("secret")
+        app_id = int(credentials.get("app_id", 0) or credentials.get("environment", {}).get("app_id", 0))
+        advertiser_id = int(credentials.get("advertiser_id", 0) or credentials.get("environment", {}).get("advertiser_id", 0))
+
```

Comment:
Could you split this parameters' parsing to 2 diff parts: parsing old and new config structure separately ?